numpy/core/tests/test_multiarray.py

```diff
@@ -1244,6 +1244,17 @@ def test_subarray_comparison(self):
         # The main importance is that it does not return True:
         with pytest.raises(TypeError):
             x == y
+        
```

Comment:
I would also slightly prefer to make a new test function here.  We have a lot of very long functions, but long test functions are not very convenient (when they fail cryptically).